bus-mapping/src/circuit_input_builder/withdrawal.rs

```diff
@@ -0,0 +1,62 @@
+//! Withdrawal & WithdrawalContext utility module.
+
+use eth_types::Address;
+
+use crate::Error;
+
+#[derive(Debug, Default)]
+/// Context of a [`Withdrawal`].
+pub struct WithdrawalContext {
+    /// Unique identifier of a withdrawal. This value starts from 0 and then increases
+    /// monotonically.
+    id: u64,
+
+    /// Reversible Write Counter tracks the number of write operations in the
+    /// call. It is incremented when there are successful writes in the sub-call of this call.
+    pub reversible_write_counter: usize,
+}
+
+impl WithdrawalContext {
+    /// Return id of this withdrawal.
+    pub fn id(&self) -> u64 {
+        self.id
+    }
+}
+
+#[derive(Debug, Copy, Clone, Default)]
+/// Result of the parsing of an Ethereum Withdrawal.
+pub struct Withdrawal {
+    /// Unique identifier of a withdrawal. This value starts from 0 and then increases
+    /// monotonically.
```

Comment:
Do we need a constraint that it starts from 0 (currently there is only for the increase by 1)?

However, it's a bit confusing because the [specs](https://github.com/privacy-scaling-explorations/zkevm-specs/blob/master/specs/withdrawal-proof.md) say: "Which means it won't be starting from 0 in most of cases."

Reply:
The id is unique and monotonical which means there is only one withdrawal with `id == 0` in all withdrawals history. For example, in block n, withdrawal id starts from 5 to 12. In block n + 1, withdrawal id will start from 13. That's why I said: "Which means it won't be starting from 0 in most of cases.".  Does it make sense to you?

Reply:
Ok, cool! Perhaps change to `Unique identifier of a withdrawal in the whole history of withdrawals.`?

Reply:
fixed in [cad4ea4](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1634/commits/cad4ea47fc78e8b5f44451124629ba6b5bdff4b4)